newIDE/app/src/AssetStore/AssetPackDialog.js

```diff
@@ -66,16 +71,31 @@ export const AssetPackDialog = ({
       setAreAssetsBeingInstalled(true);
       try {
         const installOutputs = await Promise.all(
-          assetShortHeaders.map(assetShortHeader =>
-            installAsset({
-              assetShortHeader,
-              eventsFunctionsExtensionsState,
-              project,
-              objectsContainer,
-              environment,
-            })
-          )
+          assetShortHeaders.map(async assetShortHeader => {
+            const installOutput = isPrivateAsset(assetShortHeader)
+              ? await installPrivateAsset({
+                  assetShortHeader,
+                  eventsFunctionsExtensionsState,
+                  project,
+                  objectsContainer,
+                  environment,
+                })
+              : await installPublicAsset({
+                  assetShortHeader,
+                  eventsFunctionsExtensionsState,
+                  project,
+                  objectsContainer,
+                  environment,
+                });
+
+            if (!installOutput) {
+              throw new Error('Unable to install private Asset.');
```

Comment:
The error message mentions a private asset but this can be an error linked to a public asset.

(nit: also, careful about not randomly capitalizing words in sentences.)